book/src/12_tracked.md

```diff
@@ -0,0 +1,67 @@
+# `FlaggedStorage` and modification events
+
+In most games you will have many entities, but from frame to frame only a few of them will 
+actually need to have components updated.
+
+To avoid a lot of unnecessary computation when updating components it
+would be nice if we could somehow check for only those entities that are updated
+and recalculate only those. 
+
+This is where `FlaggedStorage` comes into play. By wrapping a components
+actual storage in a `FlaggedStorage`, we can subscribe to modification events, and
+easily populate bitsets with only the entities that have actually changed.
+
+Lets look at some code:
+
+```rust,ignore
+pub struct Data {
+    [..]
+}
+
+impl Component for Data {
+    type Storage = FlaggedStorage<Self, DenseVecStorage<Self>>;
+}
+
+#[derive(Default)]
+pub struct Sys {
+    pub dirty: BitSet,
+    pub modify_reader_id: Option<ReaderId<ModifiedFlag>>,
+    pub insert_reader_id: Option<ReaderId<InsertedFlag>>,
+}
+
+impl<'a> System<'a> for Sys {
+    type SystemData = (
+        ReadStorage<'a, Data>,
+        WriteStorage<'a, SomeOtherData>,
+    );
+    
+    fn run(&mut self, (data, some_other_data): Self::SystemData) {
+        self.dirty.clear();
+        data.populate_inserted(&mut self.insert_reader_id.as_mut().unwrap(), &mut self.dirty);
+        data.populate_inserted(&mut self.modify_reader_id.as_mut().unwrap(), &mut self.dirty);
```

Comment:
Maybe add comment that you can have multiple different bitsets for different events.